### ISSUE TITLE
After update to v1.8.0, the cron schedule cannot be modified/deleted.

### ISSUE THREAD

**Issue by rockpanda** [goharbor/harbor#7868]
![定时任务报错](https://user-images.githubusercontent.com/42989547/58161408-c608d000-7cb2-11e9-8535-1c9e57b82eb4.png)
![服务状态](https://user-images.githubusercontent.com/42989547/58161410-c739fd00-7cb2-11e9-9b64-4e6ebe6d18c0.png)
[harbor-log.zip](https://github.com/goharbor/harbor/files/3206564/harbor-log.zip)

**Comment by steven-zou** [comment]
Fixed in 1.8.x

**Comment by rmoesbergen** [comment]
I'm on v1.8.1 and still seeing this issue.

**Comment by steven-zou** [comment]
@rmoesbergen 

Please provide logs (especially job service) and the dump.rdb data under `/data/redis`. We'll check.

**Comment by rmoesbergen** [comment]
In the log I see this when I try to edit the job schedule:

Jun 19 08:21:10 172.21.0.1 core[9075]: 2019-06-19T06:21:10Z [ERROR] [/common/api/base.go:229]: http error: code 500, message {"code":10006,"message":"stop job failed with error","details":"mismatch job status for stopping job: ff2cc14ca5f187b56f868157, job status Success is behind Running"}

Jun 19 08:21:10 172.21.0.1 core[9075]: 2019-06-19T06:21:10Z [ERROR] [/common/api/base.go:69]: PUT /api/system/gc/schedule failed with error: {"code":500,"message":"Internal Server Error"}

Jun 19 08:21:10 172.21.0.1 core[9075]: 2019/06/19 06:21:10 #033[1;44m[D] [server.go:2619] | 62.112.254.231|#033[41m 500 #033[0m|   4.385282ms|   match|#033[43m PUT     #033[0m /api/system/gc/schedule   r:/api/system/gc/schedule#033[0m

Does the dump.rdb file contain any secrets (e.g. should I encrypt it?)

**Comment by bovy89** [comment]
Same issue here

**Versions:**
- harbor version: 1.8.1
- docker engine version: 18.09.7
- docker-compose version: 1.23.2, build 1110ad01

@steven-zou, you got an email with dump.rdb file

**Comment by steven-zou** [comment]
@rmoesbergen / @bovy89 

What's your upgrading path?

**Comment by bovy89** [comment]
1.7.0 --> 1.8.1

**Comment by steven-zou** [comment]
> Same issue here
> 
> **Versions:**
> 
> * harbor version: 1.8.1
> * docker engine version: 18.09.7
> * docker-compose version: 1.23.2, build 1110ad01
> 
> @steven-zou, you got an email with dump.rdb file

Please also upload the logs for core and js. We need these logs to do debug.

**Comment by bovy89** [comment]
Is this enough?

```
==> jobservice.log <==
Jul 11 13:31:58 172.18.0.1 jobservice[57435]: 2019-07-11T11:31:58Z [ERROR] [/jobservice/api/handler.go:288]: Serve http request 'POST /api/v1/jobs/ebca29612c67602b085aa755' error: 500 {"code":10006,"message":"stop job failed with error","details":"mismatch job status for stopping job: ebca29612c67602b085aa755, job status Success is behind Running"}

==> core.log <==
Jul 11 13:31:58 172.18.0.1 core[57435]: 2019-07-11T11:31:58Z [ERROR] [/common/api/base.go:229]: http error: code 500, message {"code":10006,"message":"stop job failed with error","details":"mismatch job status for stopping job: ebca29612c67602b085aa755, job status Success is behind Running"}
Jul 11 13:31:58 172.18.0.1 core[57435]: 2019-07-11T11:31:58Z [ERROR] [/common/api/base.go:69]: PUT /api/system/scanAll/schedule failed with error: {"code":500,"message":"Internal Server Error"}
Jul 11 13:31:58 172.18.0.1 core[57435]: 2019/07/11 11:31:58 #033[1;44m[D] [server.go:2619] |   172.16.10.81|#033[41m 500 #033[0m|    3.19602ms|   match|#033[43m PUT     #033[0m /api/system/scanAll/schedule   r:/api/system/scanAll/schedule#033[0m
Jul 11 13:32:14 172.18.0.1 core[57435]: 2019-07-11T11:32:14Z [WARNING] [/common/config/store/driver/db.go:43]: failed to get metadata, key:reload_key, error:<nil>, skip to load item
Jul 11 13:32:14 172.18.0.1 core[57435]: 2019-07-11T11:32:14Z [WARNING] [/common/config/store/driver/db.go:43]: failed to get metadata, key:cfg_expiration, error:<nil>, skip to load item
Jul 11 13:32:14 172.18.0.1 core[57435]: 2019/07/11 11:32:14 #033[1;44m[D] [server.go:2619] |      127.0.0.1|#033[42m 200 #033[0m|   1.051692ms|   match|#033[44m GET     #033[0m /api/ping   r:/api/ping#033[0m
```

**Comment by Grounz** [comment]
Hi,

If i can help you, we have the same problem: 

- Upgrading Path: 1.7.5 to 1.8.1

**Versions**:
- harbor version: 1.8.1
- docker engine version: 18.09.3
- docker-compose version: 1.18.0, build  8dd22a9

`Jul 15 11:32:05 192.168.96.1 core[4071]: 2019-07-15T09:32:05Z [ERROR] [/common/api/base.go:229]: http error: code 500, message {"code":10006,"message":"stop job failed with error","details":"mismatch job status for stopping job: 4627031ff771a81e2b62e941, job status Success is behind Running"}`

**Comment by krisdock** [comment]
Seeing the same issue after a 1.7.5 => 1.8.1 upgrade:

from /var/log/harbor/core.log:

```
Jul 26 17:16:48 172.18.0.1 core[15136]: 2019-07-26T17:16:48Z [ERROR] [/common/api/base.go:229]: Post http://jobservice:8080/api/v1/jobs/: dial tcp 172.18.0.9:8080: getsockopt: connection refused
Jul 26 17:16:48 172.18.0.1 core[15136]: 2019-07-26T17:16:48Z [ERROR] [/common/api/base.go:69]: PUT /api/system/scanAll/schedule failed with error: {"code":500,"message":"Internal Server Error"}
```

**Comment by reasonerjt** [comment]
@steven-zou I believe we have some workarounds could you add it in the comments?

**Comment by steven-zou** [comment]
So far, there is still not a good solution to resolve this issue. We're evaluating a potential solution and check whether it will cause other problems or not. Once we decided to resolve this issue with that solution, it's probably delivered into the upcoming 1.8.3.

For a quick workaround, you can enter the Redis container and connect to database 2, then update the job status of `{harbor_job_service_namespace}:job_stats:[JOB_ID_REPLACEMENT]` to `Pending`.  Then retry to reset the schedule.

@krisdock @Grounz @bovy89 @rmoesbergen @rockpanda

**Comment by steven-zou** [comment]
@ywk253100 

As we discussed, let's apply the fix to 1.8.3 from the core side.

**Comment by ywk253100** [comment]
Fixed in 1.8.3
